mapie/utils.py

```diff
@@ -654,3 +666,369 @@ def compute_quantiles(vector: NDArray, alpha: NDArray) -> NDArray:
+
+
+def check_estimator_classification(
+        X: ArrayLike,
+        y: ArrayLike,
+        cv: Union[str, BaseCrossValidator],
+        estimator: Optional[ClassifierMixin],
+) -> ClassifierMixin:
+    """
+    Check if estimator is ``None``,
+    and returns a ``LogisticRegression`` instance if necessary.
+    If the ``cv`` attribute is ``"prefit"``,
+    check if estimator is indeed already fitted.
+    Parameters
+    ----------
+    X : ArrayLike of shape (n_samples, n_features)
+        Training data.
+    y : ArrayLike of shape (n_samples,)
+        Training labels.
+    cv : Union[str, BaseCrossValidator]
+        Cross validation parameter.
+    estimator : Optional[ClassifierMixin]
+        Estimator to check.
+    Returns
+    -------
+    ClassifierMixin
+        The estimator itself or a default ``LogisticRegression`` instance.
+    Raises
+    ------
+    ValueError
+        If the estimator is not ``None``
+        and has no fit, predict, nor predict_proba methods.
+    NotFittedError
+        If the estimator is not fitted and ``cv`` attribute is "prefit".
+    """
+    if estimator is None:
+        return LogisticRegression(multi_class="multinomial").fit(X, y)
+
+    if isinstance(estimator, Pipeline):
+        est = estimator[-1]
+    else:
+        est = estimator
+    if (
+        not hasattr(est, "fit")
+        and not hasattr(est, "predict")
+        and not hasattr(est, "predict_proba")
+    ):
+        raise ValueError(
+            "Invalid estimator. "
+            "Please provide a classifier with fit,"
+            "predict, and predict_proba methods."
+        )
+    if cv == "prefit":
+        check_is_fitted(est)
+        if not hasattr(est, "classes_"):
+            raise AttributeError(
+                "Invalid classifier. "
+                "Fitted classifier does not contain "
+                "'classes_' attribute."
+            )
+    return estimator
+
+
+def get_binning_groups(
+    y_score: NDArray,
+    num_bins: int,
+    strategy: str,
+) -> NDArray:
+    """_summary_
+    Parameters
+    ----------
+    y_score : NDArray of shape (n_samples,)
+        The scores given from the calibrator.
+    num_bins : int
+        Number of bins to make the split in the y_score.
+    strategy : string
+        The splitting strategy to split y_scores into different bins.
+    Returns
+    -------
+    NDArray of shape (num_bins,)
+        An array of all the splitting points for a new bin.
+    """
+    if strategy == "quantile":
+        quantiles = np.linspace(0, 1, num_bins)
+        bins = np.percentile(y_score, quantiles * 100)
+    if strategy == "uniform":
+        bins = np.linspace(0.0, 1.0, num_bins)
+    if strategy == "array split":
+        bin_groups = np.array_split(y_score, num_bins)
+        bins = np.sort(np.array(
+                [
+                    bin_group.max() for bin_group in bin_groups[:-1]
+                ]
+                + [np.inf]
+            )
+        )
+    return bins
```

Comment:
use elif and else

Reply:
done.

Reply:
be careful, the last else should be à elif, and there should be an else the strategy given is not allowed.

Reply:
But I already make a check a `check_split_strategy`, so this would never be tested?